StudyBuddyApp/StudyBuddyApp/Views/HelpRequestListPage.xaml.cs

```diff
@@ -84,9 +98,60 @@ private void HelpRequestListGetter()
             }).get(true);
         }
 
+
+        void Filter(string search = null, string category = null, bool own = false)
+        {
+            if (Items == null || LocalUserManager.LocalUser == null) // Dar nėra informacijos
+            {
+                return;
+            }
+            List<HelpRequestModel> filtered = Items.ToList();
+            FilteredItems.Clear();
+            filtered.ForEach((helpRequest) =>
+            {
+                if ((String.IsNullOrEmpty(category) || category == helpRequest.Category) &&
+                    (String.IsNullOrEmpty(search) || helpRequest.Title.ToLower().Contains(search) || helpRequest.Description.ToLower().Contains(search)) &&
+                    (!own || helpRequest.Name == LocalUserManager.LocalUser.Username))
+                {
+                    FilteredItems.Add(helpRequest);
+                }
+            });
+
+        }
+
         async private void ToolbarItem_Clicked(object sender, EventArgs e)
         {
-            await Navigation.PushModalAsync(new NavigationPage(new HelpRequestAddPage()));
+            AddButton.IsEnabled = false;
+            await Navigation.PushModalAsync(
+                new NavigationPage(
+                    new HelpRequestAddPage(
+                        new ViewModels.HelpRequestViewModel())));
+        }
+
+        private void RequestSearchBar_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (PickCategory.SelectedIndex >= 0)
+            {
+                Filter(RequestSearchBar.Text, Items[PickCategory.SelectedIndex].Title, false);
+            }
+            else
+            {
+                Filter(RequestSearchBar.Text, null, false);
+            }
+
+        }
+
+        private void PickCategory_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (PickCategory.SelectedIndex >= 0)
+            {
+                Filter(RequestSearchBar.Text, Items[PickCategory.SelectedIndex].Category, false);
+            }
+            else
+            {
+                Filter(RequestSearchBar.Text, null, false);
+            }
+
```

Comment:
Nėra jokio būdo grįžt prie bet kokios kategorijos (gali padaryt kad pats pirmas yra "Bet kokia" or something like that)